File: healthcare_app_server/admin_service/src/main/java/dev/skyherobrine/service/models/mongodb/ChatMessage.java
package dev.skyherobrine.service.models.mongodb;

import dev.skyherobrine.service.models.mariadb.Appointment;
import dev.skyherobrine.service.models.mariadb.Doctor;
import dev.skyherobrine.service.models.mariadb.Patient;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;
import org.springframework.data.mongodb.core.mapping.FieldType;
import org.springframework.data.mongodb.core.mapping.MongoId;

import java.time.LocalDateTime;

@Document(collection = "chat_messages")
@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
public class ChatMessage {

    @MongoId
    private Long id;
    private Appointment appointment;
    private String message;
    @Field(name = "is_recall")
    private Boolean isRecall;
    private ChatMessage reply;
    @Field(name = "created_at", targetType = FieldType.DATE_TIME)
    private LocalDateTime createdAt;
    @Field(name = "updated-at", targetType = FieldType.DATE_TIME)
    private LocalDateTime updatedAt;
}

File: healthcare_app_server/chat_service/models/chat-message.js
class ChatMessage {
    constructor(id, roomCall, reply, isRecall, createdAt, updatedAt) {
        this._id = id;
        this._roomCall = roomCall;
        this._reply = reply;
        this._isRecall = isRecall;
        this._createdAt = createdAt;
        this._updatedAt = updatedAt;
        this._appointment = appointment;
    }

    get id() {
        return this._id;
    }

    set id(value) {
        this._id = value;
    }

    get roomCall() {
        return this._roomCall;
    }

    set roomCall(value) {
        this._roomCall = value;
    }

    get reply() {
        return this._reply;
    }

    set reply(value) {
        this._reply = value;
    }

    get isRecall() {
        return this._isRecall;
    }

    set isRecall(value) {
        this._isRecall = value;
    }

    get createdAt() {
        return this._createdAt;
    }

    set createdAt(value) {
        this._createdAt = value;
    }

    get updatedAt() {
        return this._updatedAt;
    }

    set updatedAt(value) {
        this._updatedAt = value;
    }
}

module.exports = ChatMessage